lib/async.js

```diff
@@ -75,6 +77,7 @@ module.exports = function resolve(x, options, callback) {
     var extensions = opts.extensions || ['.js'];
     var basedir = opts.basedir || path.dirname(caller());
     var parent = opts.filename || basedir;
+    var conditions = opts.ignoreExportsField === false ? ['require', 'node'] : [];
```

Comment:
I get the idea of `auto`, but what about conditional exports (added in 13.7) and star exports (added to node after this PR was created)?
Would these also be enabled only if the node version supports it or is exports an all-or-nothing behaviour?

Reply:
https://www.npmjs.com/package/has-package-exports `has-package-exports/conditional` can detect the first one, and I'll add an entry point to that package to detect star exports - i'll also make a new package for package imports :-)

That's a good question - which leads me to think that would be *ideal* is that we'd support more than just "ignore" and "respect", but also add an additional mode for each permutation ("respect, pre-conditions" and "respect, without stars") and then "respect" would include whatever latest node did.

That then suggests that we should do "auto" in a future PR, and *not* in this one. However, that does raise the question about how we should do "exports without stars" - as "respect" in this PR, or make this PR do "respect without stars" and add "respect" in a future PR?

Reply:
> That then suggests that we should do "auto" in a future PR, and not in this one. However, that does raise the question about how we should do "exports without stars" - as "respect" in this PR, or make this PR do "respect without stars" and add "respect" in a future PR?

I'd argue we should have both at the moment but make the two options effectively the same. When support for star exports gets added, `respect` behaviour would change and `respect, without stars` would not.

Reply:
Thinking a bit more on it I wonder if there's a usecase for all of these options 🤔 

`respect`: definitely useful
`ignore`: also definitely useful
`auto`: arguably useful

However, for the distinctions between "respect", "respect, without stars" and "respect, without conditions" it's a lot less clear whether this would be useful.
I can't think of a situation where I'd explicitly want to enable exports but not conditions, or exports but not stars.

And it's not just that I doubt the usefulness, I think these are counterintuitive and counterproductive.
A developer passing in `'ignore'` is on their own. They've chosen not to follow exports defined by a package so that's what they'll get.
A developer passing in `'respect, without stars'` is asking for partial support of exports. This is not something that I as a package author necessarily took into account when writing my package, especially not if my package's engine field requires a minimum node version that supports stars exports.

The same doesn't go for `'auto'`. I don't see an issue with `'auto'` resulting in `'respect, without stars'` behaviour when in a node version that doesn't support stars exports. In my mind `auto` means: mimic the behaviour of the current node version, so if I'm running node 12 I'll get export of node 12 just like I would get with `require.resolve`.
In the example above, where I wrote a package that uses stars exports and defined the engines field to require a minimum node version that supports it, this should not give problems assuming package managers warn/error on installing my package on an older node version.

Reply:
The use case is being able to represent the resolution algorithm for specific versions of node, not just the latest version.

As a package author, if your package supports 12.17, for example, you *should* be checking that there's no star patterns, since those are unsupported.

Specifically, any mode "auto" picks must also be able to be explicitly specified, otherwise you'd have to switch node versions to be able to test it.

Reply:
Testing makes a lot of sense, I should've thought of that 😅